services/venice-router/src/main/java/com/linkedin/venice/router/api/path/VeniceMultiKeyPath.java

```diff
@@ -230,14 +240,19 @@ public int getLongTailRetryMaxRouteForMultiKeyReq() {
 
   @Override
   public boolean isLongTailRetryAllowedForNewRoute() {
+    // We are keeping both features for now. Retry is only allowed if both conditions pass:
+    // 1. retry attempts < longTailRetryMaxRouteForMultiKeyReq (if set).
+    // 2. we still have retry budget according to retry manager.
+    boolean longTailRetryAllowed;
     if (longTailRetryMaxRouteForMultiKeyReq == -1) {
       // feature is disabled
-      return true;
-    }
-    if (longTailRetryMaxRouteForMultiKeyReq <= 0) {
-      return false;
+      longTailRetryAllowed = true;
```

Comment:
nit: can we just `return retryManager.isRetryAllowed();` here
same for next if, `return false` ..

Reply:
No, rather removing `longTailRetryAllowed`

Reply:
Oh I see, my bad. Done.